misc/ls-extensions/modules/metadata-to-record-generator/src/main/java/io/ballerina/generators/MetadataToRecordSchemaRequest.java

```diff
@@ -0,0 +1,50 @@
+package io.ballerina.generators;
+
+/**
+ *  Request format for customRecordGenerator endpoint.
+ */
+public class MetadataToRecordSchemaRequest {
+    private String metadata;
+    private String schemaSource;
+    private boolean isClosed;
+    private boolean isRecordTypeDesc;
+
+    public MetadataToRecordSchemaRequest(String jsonString, String schemaSource, boolean isClosed) {
```

Comment:
Let's add a more meaningful name for `jsonString` parameter like `metadataJsonString`